Add Force update to edit service if service is in deployment

Poltergeist opened the pull request.

mesosphere-web commented:
Reviewer: @jfurrow

jfurrow commented on src/js/components/modals/ServiceFormModal.js:
```diff
@@ -209,6 +210,7 @@ class ServiceFormModal extends mixin(StoreMixin) {
 
   onMarathonStoreServiceEditError(errorMessage) {
     this.setState({
+      force: errorMessage.message && /force=true/.test(errorMessage.message),
```
What do you think about putting this `/force=true/.test...` check in a separate method on the `ServiceFormModal`? Something like `#shouldForceEdit` that accepts the `message` and returns `/force=true/.test(message)`.

Poltergeist replied:
Comment has been addressed.

jfurrow commented:
One minor comment, but this looks good!

rcorral commented on src/js/events/MarathonActions.js:
```diff
@@ -89,9 +89,15 @@ module.exports = {
+    let url = `${Config.rootUrl}${Config.marathonAPIPrefix}/apps/${data.id}`;
+
+    if (force === true) {
+      url += '?force=true';
```
It would be good to add a simple unit test to make sure that the `force` argument changes the requested URL.

MatApple commented:
Hey @Poltergeist can you post a screenshot of integration tests passing? After that, and Rafas comment I think this looks good!

pierlo-upitup commented on src/js/components/modals/ServiceFormModal.js:
```diff
@@ -205,13 +206,18 @@ class ServiceFormModal extends mixin(StoreMixin) {
     });
   }
 
+  shouldForceUpdate(message = this.errorMessage) {
```
Probably unrelated, but why aren't we storing `errorMessage` in this.state?

Poltergeist replied:
shouldForceUpdate is called in the setStates. This is why this function takes a parameter. But in fact the default is wrong. Changed it.

Poltergeist commented:
![image](https://cloud.githubusercontent.com/assets/156010/16268597/55c902a2-388f-11e6-83ba-3d33a568b186.png)

![image](https://cloud.githubusercontent.com/assets/156010/16268589/4db30248-388f-11e6-9db0-1645ab35041f.png)

Added a unit test and fixed the integration tests.
